### PR TITLE
added LOGIN client

### PR DESCRIPTION
**Pull request by wildart**
This will allow to connect to services which support LOGIN method, e.g. office365.com

### CODE REVIEW

**Comment by emersion** [comment]
Any reason why you can't use PLAIN instead? LOGIN is deprecated.

**Comment by wildart** [comment]
Some Outlook Exchange SMTP servers and smtp.office365.com, that I use, only accept LOGIN method. [So, I cannot use `aerc` to send anything](https://todo.sr.ht/~sircmpwn/aerc2/263).

**Comment by wildart** [comment]
Yep, MS Exchange SMTP servers use AUTH LOGIN, see [MS-XLOGIN](https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-xlogin/e410ec76-ab45-47fb-a49c-b557d99e671f). There is no other way to connect to them.

**Comment by emersion** [comment]
Just to be sure, can you post the output of `openssl s_client -connect server:port`?

If there's no other solution, can you document `NewLoginClient` and make it clear that it's deprecated?

**Comment by wildart** [comment]
```
[~]$ openssl s_client -connect smtp.office365.com:587                                                                                                          
CONNECTED(00000003)
140461130774168:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:s23_clnt.c:794:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 305 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1565902424
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
```

**Comment by emersion** [comment]
For future reference, the right command is `openssl s_client -connect smtp.office365.com:587 -starttls smtp -crlf`. The server capabilities are:

```
EHLO localhost
250-HE1P190CA0059.outlook.office365.com Hello [83.245.224.167]
250-SIZE 157286400
250-PIPELINING
250-DSN
250-ENHANCEDSTATUSCODES
250-AUTH LOGIN XOAUTH2
250-8BITMIME
250-BINARYMIME
250-CHUNKING
250 SMTPUTF8
```

So yeah, PLAIN is missing. Thanks Microsoft.

File: login.go [hunk 1]
```diff
@@ -1,8 +1,43 @@
 package sasl
 
+import (
+	"bytes"
+)
+
 // The LOGIN mechanism name.
 const Login = "LOGIN"
 
+var expectedChallenge = []byte{80, 97, 115, 115, 119, 111, 114, 100, 58}
```
Comment (emersion) [inline login.go]: `[]byte("Password:")` would be more readable.

File: login.go [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
+	if bytes.Compare(challenge, expectedChallenge) != 0 {
+		return nil, ErrUnexpectedServerChallenge
+	} else {
+		return []byte(a.Password), nil
+	}
+}
+
+// A client implementation of the LOGIN authentication mechanism for SMTP,
+// as described in
+// https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-xlogin/
```
Comment (emersion) [inline login.go]: Maybe reference the RFC draft instead?

File: login.go [hunk 3]
```diff
@@ -1,8 +1,43 @@
...
+//
+// This mechanism is used by Microsoft Exchange SMTP servers. It is considered
+// obsolete, and should not be used when other mechanisms are available.
```
Comment (emersion) [inline login.go]: We could add "For plaintext password authentication, the standard mechanism is PLAIN".

File: login.go (hunk 2 above)
Comment (wildart) [reply login.go]: I'm not sure that there is some kind of RFC for it. I found this: https://datatracker.ietf.org/doc/draft-murchison-sasl-login/, but it is way too old.

File: login.go (hunk 3 above)
Comment (wildart) [reply login.go]: Added.

File: login.go (hunk 2 above)
Comment (emersion) [reply login.go]: [IANA](https://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml) lists the RFC draft, so I'd prefer referring to it as well. The "updated" Microsoft PDF doesn't change anything anyway.

**Pull request merged**